akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/PersistentFsmToTypedMigrationSpec.scala

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.akka.persistence.typed
+
+import java.util.UUID
+
+import akka.actor.PoisonPill
+import akka.actor.testkit.typed.scaladsl.ActorTestKit
+import akka.actor.typed.ActorRef
+import akka.persistence.fsm.PersistentFSM.StateChangeEvent
+import akka.persistence.fsm.PersistentFSMSpec.DomainEvent
+import akka.persistence.fsm.PersistentFSMSpec.EmptyShoppingCart
+import akka.persistence.fsm.PersistentFSMSpec.Item
+import akka.persistence.fsm.PersistentFSMSpec.ItemAdded
+import akka.persistence.fsm.PersistentFSMSpec.OrderDiscarded
+import akka.persistence.fsm.PersistentFSMSpec.OrderExecuted
+import akka.persistence.fsm.PersistentFSMSpec.ShoppingCart
+import akka.persistence.typed.{ EventAdapter, EventSeq, ExpectingReply, PersistenceId, SnapshotAdapter }
+import akka.persistence.typed.scaladsl.Effect
+import akka.persistence.typed.scaladsl.EventSourcedBehavior
+import akka.persistence.typed.scaladsl.PersistentFSMMigration
+import com.typesafe.config.ConfigFactory
+import org.scalatest.WordSpec
+import org.scalatest.concurrent.ScalaFutures
+
+object PersistentFsmToTypedMigrationSpec {
+  val config = ConfigFactory.parseString(s"""
+    akka.persistence.journal.leveldb.dir = "target/typed-persistence-${UUID.randomUUID().toString}"
+    akka.persistence.journal.plugin = "akka.persistence.journal.leveldb"
+    akka.persistence.snapshot-store.plugin = "akka.persistence.snapshot-store.local"
+    akka.persistence.snapshot-store.local.dir = "target/typed-persistence-${UUID.randomUUID().toString}"
+    """)
+
+}
+
+object ShoppingCartActor {
```

Comment:
ShoppingCartBehavior?

bonus is that we might be able to use a variant of this sample for ticket #25473